xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultQuoteService.java

```diff
@@ -0,0 +1,124 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.mentions.internal;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.rendering.block.MacroBlock;
+import org.xwiki.rendering.block.ParagraphBlock;
+import org.xwiki.rendering.block.XDOM;
+import org.xwiki.rendering.block.match.BlockMatcher;
+import org.xwiki.rendering.renderer.BlockRenderer;
+import org.xwiki.rendering.renderer.printer.DefaultWikiPrinter;
+import org.xwiki.rendering.renderer.printer.WikiPrinter;
+import org.xwiki.text.StringUtils;
+
+import static org.xwiki.rendering.block.Block.Axes.DESCENDANT;
+
+/**
+ * Default implementation of {@link QuoteService}.
+ *
+ * @version $Id$
+ * @since 12.6RC1
+ */
+@Component
+@Singleton
+public class DefaultQuoteService implements QuoteService
+{
+    @Inject
+    @Named("plainmentions/1.0")
+    private BlockRenderer renderer;
+
+    @Override
+    public Optional<String> extract(XDOM xdom, String anchorId)
+    {
+        List<Block> blockList = xdom.getBlocks(new ExactAnchorBlockMatcher(anchorId), DESCENDANT);
+
+        Optional<Block> firstBlock;
+        // Best effort selection of the mentions. In case of ambiguity we do not generation a quote 
+        // for the provided anchorId.
+        // Note that if xdoom contains a single empty anchor, it is not ambiguous and a quote will
+        // be extracted.
+        if (blockList.size() == 1) {
+            firstBlock = Optional.of(blockList.get(0));
+        } else {
+            firstBlock = Optional.empty();
+        }
+
+        return firstBlock
+                   .map(new RenderFunction())
+                   .map(it -> StringUtils.abbreviate(it, "...", 200));
+    }
+
+    private static class ExactAnchorBlockMatcher implements BlockMatcher
+    {
+        private final String anchorId;
+
+        ExactAnchorBlockMatcher(String anchorId)
+        {
+            this.anchorId = anchorId;
+        }
+
+        @Override
+        public boolean match(Block block)
+        {
+            boolean ret;
+            if (block instanceof MacroBlock) {
+                ret =
+                    ((MacroBlock) block).getId().equals("mention")
+                        && Optional.ofNullable(block.getParameter("anchor"))
+                               .map(it -> it.equals(this.anchorId))
+                               .orElse(false);
```

Comment:
sounds to me that if the anchor parameter is null you keep returning false here.